### DISCUSSION TITLE
Best practices for fetching multiple ENS names/avatars?

### DISCUSSION THREAD

**Discussion by JamieLottering** [wevm/wagmi#2133] [Q&A]
I have a table component that has a list of addresses. I want to fetch the ens name and avatar for each address. By using `useEnsName` and `useEnsAvatar` in each row component, it makes a LOT of requests to alchemy and we get rate limited. Is there a way to batch these requests, or some best practice around this?

![image](https://user-images.githubusercontent.com/134010/230161763-bcc5ea77-6b50-4929-a727-32655e7c92d7.png)

**Comment by JamieLottering** [comment]
Just saw that viem is supporting multicalls https://twitter.com/jakemoxey/status/1648805038159204352?s=20 . Is it on the roadmap to support batched calls in wagmi as well?

**Reply by jxom** [reply]
Yep. Whatever viem has, wagmi will also have. 😉

**Reply by jxom** [reply]
https://github.com/wagmi-dev/wagmi/discussions/2207
